recipes/sbp/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+import os
+
+from textwrap import dedent
+
+import yaml
+
+from conans import ConanFile, CMake, tools
+
+
+class SbpConan(ConanFile):
+    name = "sbp"
+    license = "MIT"
+    author = "Swift Navigation <dev@swiftnav.com>"
+    url = "https://github.com/swift-nav/libsbp"
+    description = "Swift Binary Protocol client library"
+    topics = ("gnss",)
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": True, "fPIC": True}
+    generators = "cmake"
+    generators = "cmake"
+    exports = "submoduledata.yml"
+    exports_sources = "c"
+
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self._source_subfolder
+        )
+
+        submodule_filename = os.path.join(self.recipe_folder, "submoduledata.yml")
```

Comment:
Instead of using `submoduledata.yml`, you can also add multiple items in `conandata.yml` and extract both of them:
```yml
sources:
  "3.4.10":
    - url: "https://github.com/swift-nav/libsbp/archive/refs/tags/v3.4.10.tar.gz"
      sha256: "f2fb738f49112b25e7849ca0c75415159127e9b5373b13e7027362b8fa0b1224"
    - url: "https://github.com/swift-nav/cmake/archive/373d4fcafbbc0c208dc9ecb278d36ed8c9448eda.zip"
      sha256: "6077d2a754d013e3cb9826f589e47b19ab01f4d91ede4f5bfc14db74bc5dc894"
```
An do the following in `source`:
```python
        tools.get(**self.conan_data["sources"][self.version][0],
                  strip_root=True, destination=self._source_subfolder)
        tools.get(**self.conan_data["sources"][self.version][1],
                  strip_root=True, destination=os.path.join(self._source_subfolder, "c", "cmake", "common")
```

Reply:
I noticed `submoduledata.yml` in two other packages on conan-center, namely `yojimbo` and `open62541`.

As this file can be generated from the git repo, it seems to be the more appropriate file format.  I'd be looking to also re-use this approach for the other swift-nav projects, which all currently heavily use gitmodules.  e.g. https://github.com/swift-nav/gnss-converters/blob/master/.gitmodules which defines the shas needed from https://github.com/swift-nav/libubx and https://github.com/swift-nav/libixcom which have no tags/versions.

I'm happy to switch to your suggested approach if you still prefer that approach.  I'll also investigating whether the swift-nav cmake repo can be versioned as I've seen we can have conan packages which provide cmake modules, so that could be a way to supply this dep of all the swift-nav tools.

Reply:
[gnss-converter](https://github.com/swift-nav/gnss-converters/blob/master/.gitmodules) is a bad example.
All submodules are external projects that should be provided by conan packages instead of vendored built-ins.
Creating a conan recipe for libubx + libixcom and adding it as a requirement is *the conan way*.

Reply:
I would say all urls in conandata.yml like cryptopp recipe:
https://github.com/conan-io/conan-center-index/blob/2a0e7d5f7f71f9af18efe270ba068e258e3cdb51/recipes/cryptopp/all/conandata.yml

Does the reducer properly exclude others versions in submoduledata.yml?